lib/results_fuzz.go

```diff
@@ -0,0 +1,46 @@
+// +build gofuzz
+
+package vegeta
+
+import (
+	"bytes"
+	"io"
+)
+
+// FuzzResultsFormatDetection tests result list format detection.
+func FuzzResultsFormatDetection(fuzz []byte) int {
+	DecoderFor(bytes.NewReader(fuzz))
```

Comment:
Should we also test that we can read all results from the detected decoder?

Reply:
Good idea. Added that reading.